examples/js/lines/LineSegments2.js

```diff
@@ -52,6 +52,142 @@ THREE.LineSegments2.prototype = Object.assign( Object.create( THREE.Mesh.prototy
 
 		};
 
-	}() )
+	}() ),
+
+	raycast: ( function () {
+
+		var start = new THREE.Vector4();
+		var end = new THREE.Vector4();
+
+		var ssOrigin = new THREE.Vector4();
+		var ssOrigin3 = new THREE.Vector3();
+		var mvMatrix = new THREE.Matrix4();
+		var line = new THREE.Line3();
+		var closestPoint = new THREE.Vector3();
+
+		return function raycast( raycaster, intersects ) {
+
+			if ( raycaster.camera === null ) {
+
+				console.error( 'LineSegments2: "Raycaster.camera" needs to be set in order to raycast against LineSegments2.' );
+
+			}
+
+			var ray = raycaster.ray;
+			var camera = raycaster.camera;
+			var projectionMatrix = camera.projectionMatrix;
+
+			var geometry = this.geometry;
+			var material = this.material;
+			var resolution = material.resolution;
+			var lineWidth = material.linewidth;
+
+			var instanceStart = geometry.attributes.instanceStart;
+			var instanceEnd = geometry.attributes.instanceEnd;
+
+			// ndc space [ - 1.0, 1.0 ]
+			ray.at( 1, ssOrigin );
```

Comment:
This is set to get some point out along the distance of the ray so that the origin of the camera which makes `ssOrigin.w` 0 after applying the projection matrix. I can add a comment unless there's a different preferred solution.

Reply:
Yes, a reasonably-detailed comment would be nice.